### PR TITLE
feat(unlock-protocol-com) - add upcoming events page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
This PR adds "upcoming events" page in the main website, to list all future events

- [x] add template for events 
- [x] add upcoming events page
- [x] add link in navbar
- [x] get events from ics file and shows it 
- [x] add link to add event in calendar
- [x] add styling for mobile 
- [x] sort events by date asc / sort past event desc
- [x] show event date with the correct time for different timezons  

# Issues

Fixes #
Refs #

**Navbar**

Screenshot 2022-07-27 at 18 37 19

**Show only future events**

Screenshot 2022-07-28 at 11 41 40

**Show only future events**

Screenshot 2022-07-28 at 11 44 51

**Resposive view**

Screenshot 2022-07-28 at 13 19 39

**Add to calendar view**

Screenshot 2022-07-28 at 13 10 34

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-protocol-com/src/components/pages/Events/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            our profile to join the conversation!
+            https://twitter.com/UnlockProtocol
+          </span>
+          <div className="flex gap-[8px] items-center">
+            <MapPinIcon />
+            <span className="text-brand-ui-primary text-lg font-bold">
+              https://google.it
+            </span>
+          </div>
+          <Button variant="secondary">Add to Calendar</Button>
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-protocol-com/src/components/pages/Events/index.tsx:41]: Oh yes, is secondary in design

[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-protocol-com/src/components/pages/Events/index.tsx]: Added component

[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-protocol-com/src/components/pages/Events/index.tsx]: changed with `primary`

**Comment by julien51** [comment]
@kalidiagne It looks like events are not sorted?
I think it would be nice to show _all_ events but the ones in the past are slightly greyed out and the button disabled? wdyt @ChialiTsai ?

File: unlock-protocol-com/src/components/pages/Events/index.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+    <div className="mx-auto max-w-7xl px-6 lg:px-0">
+      <header className="flex flex-col gap-[16px] pt-[24px]">
+        <h1 className="heading">Upcoming Events</h1>
+        <p className="text-lg">
+          Join us for AMA, Governance and other activities
+        </p>
+      </header>
+
+      <section className="grid grid-cols-1 gap-[32px] py-[40px] lg:grid-cols-3">
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-protocol-com/src/components/pages/Events/index.tsx]: yep! changed with the default one!

**Comment by kalidiagne** [comment]
> @kalidiagne It looks like events are not sorted? I think it would be nice to show _all_ events but the ones in the past are slightly greyed out and the button disabled? wdyt @ChialiTsai ?

Sorted events and added link to calendar (screen attached), waiting for @ChialiTsai feedback!

**Comment by julien51** [comment]
@kalidiagne : 
> the ones in the past are slightly greyed out and the button disabled? wdyt @ChialiTsai ?

File: unlock-protocol-com/tailwind.config.js [hunk 3]
```diff
@@ -4,4 +4,5 @@ module.exports = {
     './../node_modules/@unlock-protocol/ui/dist/*.{js,css}',
   ],
   presets: [require('@unlock-protocol/ui/dist/unlock-tailwind-preset')],
+  plugins: [require('@tailwindcss/line-clamp')],
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-protocol-com/tailwind.config.js:7]: awesome! i'll remove

**Comment by kalidiagne** [comment]
**Future and past events**

Screenshot 2022-07-28 at 17 22 49

**Comment by kalidiagne** [comment]
what about ignoring events without description and location as this example? 

Screenshot 2022-07-28 at 17 25 35

@julien51 @ChialiTsai

**Comment by kalidiagne** [comment]
**Placeholder**

Screenshot 2022-07-28 at 18 35 56

**Comment by ChialiTsai** [comment]
image



@kalidiagne is it possible to hide the past event add to calendar btn?

**Comment by kalidiagne** [comment]
> image
> 
> @kalidiagne is it possible to hide the past event add to calendar btn?

sure! done!

File: unlock-protocol-com/src/utils/calendar.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+  })
+}
+
+export const icalEventsToJson = async (
+  fileUrl: string,
+  type: 'future' | 'past' | 'all' // get future or past events or all
+): Promise<CalendarEvent[]> => {
+  try {
+    const calendarRaw = await fetch(fileUrl).then((res) => res.text())
```
Comment (kalidiagne) [inline unlock-protocol-com/src/utils/calendar.ts:72]: getting data from url 

`https://calendar.google.com/calendar/ical/c_dm3gud9ph6jqk7epq6jtij8u00%40group.calendar.google.com/public/basic.ics`
actually from localhost bypassing with cors plugin gonna work online? 
@searchableguy

**Comment by julien51** [comment]
@kalidiagne I tested and it does not work https://62e2c9dc311c8f0068b12403--unlock-protocol-com-staging.netlify.app/upcoming-events

You need to generate the page content at build time.... so that you don't have CORS issues.

File: unlock-protocol-com/package.json [hunk 5]
```diff
@@ -25,6 +25,8 @@
     "babel-plugin-require-context-hook": "1.0.0",
     "babel-plugin-styled-components": "2.0.7",
     "babel-preset-react-app": "10.0.1",
+    "cal-parser": "^1.0.2",
```
Comment (julien51) [inline unlock-protocol-com/package.json:28]: Remove carets please!

**Comment by kalidiagne** [comment]
Updated and checked also that event time renders with timezone time

Screenshot 2022-07-29 at 11 35 11

Screenshot 2022-07-29 at 11 34 41


Screenshot 2022-07-29 at 11 35 24

File: unlock-protocol-com/src/config/routes.tsx [hunk 6]
```diff
@@ -83,4 +83,12 @@ export const routes: Routes = {
         'Apply for Unlock Grant Program to receive UDT tokens, technical and promotional support.',
     }),
   },
+  upcomingEvents: {
+    label: 'Upcoming Events',
+    seo: customizeSEO({
+      path: '/upcoming-events',
+      title: 'Upcoming Events',
+      description: 'Join us for AMA, Governance and other activities ',
```
Comment (julien51) [inline unlock-protocol-com/src/config/routes.tsx:91]:
```suggestion
      description: 'Join the Unlock Protocol community and core teams for conferences, workshops, AMA, Governance and, other activities',
```

**Review by julien51 — CHANGES_REQUESTED** [review]
LG, please test in the draft built (especially the links...)

File: unlock-protocol-com/package.json (hunk 5 above)
Comment (julien51) [reply unlock-protocol-com/package.json:28]: (and run `yarn`)

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
Woot! LGTM! Let's merge please!

**Pull request merged**